### DISCUSSION TITLE
Saturation Issue with NeumannBC in Cahn-Hilliard

### DISCUSSION THREAD

**Discussion by ashwinshekhar21** [idaholab/moose#20308] [Q&A Modules: Phase field] [answered]
Hello,

We are attempting to modify the [tutorial code](https://mooseframework.inl.gov/modules/phase_field/Tutorial.html) for the spinodal decomposition of the Cr-Fe alloy provided with MOOSE. We start with a somewhat homogeneous initial concentration of a value less than that required for spinodal decomposition. 

Our objective is to induce the spinodal decomposition locally, by imposing a flux on the left boundary, such that the concentration gradually shifts towards the right in the double-well delG vs concentration curve, inducing the decomposition into an advancing front of c=0.8 Cr followed by the rest being ~0.2 Cr.

![image](https://user-images.githubusercontent.com/47918128/154125941-4a85c430-beb3-46d3-bed7-edbbeb510295.png)

Over time, we expected one of the decomposition product phases (orange in our output, Cr mol = ~0.8), to grow and progressively occupy a larger fraction. However, it saturates after a certain number of timesteps and it is not clear why this happens. For instance, here, this configuration is attained relatively early in the simulation, and nothing happens until the simulation ends:

https://user-images.githubusercontent.com/47918128/154202112-4f03419a-8210-494b-93b7-81212239a116.mp4

This does not appear to depend on the length of the simulation either (same result even if we run for t=280 weeks etc.). We are implementing periodic boundary conditions in the y-direction, and imposing NeumannBC value = 1e-5 on the left and = 0 on the right. Our initial concentration is approximately 0.35 mol Cr (RandomIC, variation from 0.34-0.36). 

Here is our code:

```
[Mesh]
  type = GeneratedMesh
  dim = 2
  elem_type = QUAD4
  nx = 25
  ny = 25
  nz = 0
  xmin = 0
  xmax = 25
  ymin = 0
  ymax = 25
  zmin = 0
  zmax = 0
  uniform_refine = 2
[]

[Variables]
  [./c]   # Mole fraction of Cr (unitless)
    order = FIRST
    family = LAGRANGE
  [../]
  [./w]   # Chemical potential (eV/mol)
    order = FIRST
    family = LAGRANGE
  [../]
[]

[ICs]
  [./concentrationIC]   # 35 wt% Cr with variations
    type = RandomIC
    min = 0.34
    max = 0.36
    seed = 210
    variable = c
  [../]
[]

[BCs]
  [./Periodic]
    [./c_bcs]
      auto_direction = 'y'
    [../]
  [../]
  [./left]
    type = NeumannBC
    variable = c
    boundary = left
    value = 1e-5
  [../]
  [./left]
    type = NeumannBC
    variable = c
    boundary = right
    value = 0
  [../]
[]

[Kernels]
  [./w_dot]
    variable = w
    v = c
    type = CoupledTimeDerivative
  [../]
  [./coupled_res]
    variable = w
    type = SplitCHWRes
    mob_name = M
  [../]
  [./coupled_parsed]
    variable = c
    type = SplitCHParsed
    f_name = f_loc
    kappa_name = kappa_c
    w = w
  [../]
[]

[Materials]
  [./constants]
    type = GenericFunctionMaterial
    prop_names = 'kappa_c M'
    prop_values = '8.125e-16*6.24150934e+18*1e+09^2*1e-27
                   2.2841e-26*1e+09^2/6.24150934e+18/1e-27'
                   # kappa_c*eV_J*nm_m^2*d
                   # M*nm_m^2/eV_J/d
  [../]
  [./local_energy]
    type = DerivativeParsedMaterial
    f_name = f_loc
    args = c
    constant_names = 'A   B   C   D   E   F   G  eV_J  d'
    constant_expressions = '-2.446831e+04 -2.827533e+04 4.167994e+03 7.052907e+03
                            1.208993e+04 2.568625e+03 -2.354293e+03
                            6.24150934e+18 1e-27'
    function = 'eV_J*d*(A*c+B*(1-c)+C*c*log(c)+D*(1-c)*log(1-c)+
                E*c*(1-c)+F*c*(1-c)*(2*c-1)+G*c*(1-c)*(2*c-1)^2)'
    derivative_order = 2
  [../]
[]

[Postprocessors]
  [./step_size]          
    type = TimestepSize
  [../]
  [./iterations]          
    type = NumNonlinearIterations
  [../]
  [./nodes]                
    type = NumNodes
  [../]
  [./evaluations]           
    type = NumResidualEvaluations
  [../]
  [./active_time]          
    type = PerfGraphData
    section_name = "Root"
    data_type = total
  [../]
[]

[Preconditioning]
  [./coupled]
    type = SMP
    full = true
  [../]
[]

[Executioner]
  type = Transient
  solve_type = NEWTON
  l_max_its = 30
  l_tol = 1e-6
  nl_max_its = 50
  nl_abs_tol = 1e-9
  end_time = 16934400   # 7*28 days
  petsc_options_iname = '-pc_type -ksp_gmres_restart -sub_ksp_type
                         -sub_pc_type -pc_asm_overlap'
  petsc_options_value = 'asm      31                  preonly
                         ilu          1'
  [./TimeStepper]
    type = IterationAdaptiveDT
    dt = 10
    cutback_factor = 0.8
    growth_factor = 1.5
    optimal_iterations = 7
  [../]
  [./Adaptivity]
    coarsen_fraction = 0.1
    refine_fraction = 0.7
    max_h_level = 2
  [../]
[]

[Debug]
  show_var_residual_norms = true
[]

[Outputs]
  exodus = true
  console = true
  [./console]
    type = Console
    max_rows = 10
  [../]
[]
```

**_Is there any reason for this apparent "saturation", and is there any way to rectify it such that the transition continues to grow? Such that the orange portion in our simulation box eventually occupies the entire area?_**

Thank you!

**Comment by ykvishal** [comment, marked as answer]
Interesting issue.

If you replace your initial condition by following then what will happen?

```
[ICs]
  [./concentrationIC] 
     type = BoundingBoxIC
     x1 = 0
     y1 = 0
     x2 = 20
     y2 = 25
     inside = 0.823659
     outside = 0.236248
    variable = c
  [../]
[]
```
you can vary x2 from 5 to 20 and comment out  `end_time = 16934400`

**Reply by ashwinshekhar21** [reply]
Thank you for your response.

So we tried out the above modifications, as suggested. The resulting output is as follows (showing here for x2 = 15, although this is emblematic, as all the x2 values result in similar behaviour):

https://user-images.githubusercontent.com/47918128/154244205-d83fd8ea-0664-47d5-9c00-4944c9b6db65.mp4

The relative fraction of the two phases remains unchanged. The only observable change is the establishment of a diffuse interface, but that's to be expected.

**Reply by ykvishal** [reply]
This suggests that interface won't move after reaching to equilibrium concertation from common tangent construction. It seems that MOOSE is doing fine. I may be wrong, but the issue is related to your problem definition. If you change value = 1e-5 to value = 1e-3 or value = 1e-1 then it may not converge after reaching equilibrium concentration.
